# 7.3 L, L over D, K, and Kv for Conventional Type Valves

A 100 mm class 600 steel angle valve, has a full area seat.

Calculate its resistance coefficient K, flow coefficient Kv, the equivalent length of it L/D, and the length for complete turbulent L.

In [1]:
from fluids.units import *
NPS, Di, Do, t = nearest_pipe(Do=0.103*u.m, schedule='80')
fd = 0.0165 # provided - note equivalent length is proportional to this value
d = 0.0972*u.m # diameter of seat
K = K_angle_valve_Crane(D1=d, D2=Di, fd=fd, style=1)
Kv = K_to_Kv(K, d)
L_D = L_equiv_from_K(K, fd)
L = L_D*d

print('Loss coefficient = %s' %K)
print('Valve flow coefficient = %s' %Kv)
print('Equivalent length = %s' % L_D)
print('Length for complete turbulence = %s' %L)

Loss coefficient = 2.4729637821072443 dimensionless
Valve flow coefficient = 240.31652930826877 meter ** 3 / hour
Equivalent length = 149.8765928549845 dimensionless
Length for complete turbulence = 14.568004825504492 meter


The values given in the solution are K=2.475, Kv = 240.2, L/D=150, and L = 14.6.